OptimizelySDK/Event/Internals/UserEventFactory.cs

```diff
@@ -0,0 +1,98 @@
+﻿using OptimizelySDK.Entity;
+using OptimizelySDK.Event.Entity;
+
+namespace OptimizelySDK.Event.internals
+{
+    /// <summary>
+    /// UserEventFactory builds ImpressionEvent and ConversionEvent objects from a given UserEvent.
+    /// </summary>
+    public class UserEventFactory
```

Comment:
Is there a strong reason for making this a separate file? I was thinking a generic "Event" factory can create all these types of events

Reply:
I think it's a categorization between userevents which are impression and conversion and server event which is only LogEvent. Same in java. that's why I did it.

Reply:
Ok that is fine then.